plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/TableStatisticsMaker.java

```diff
@@ -244,6 +241,29 @@ private NullableValue makeNullableValue(io.trino.spi.type.Type type, Object valu
         return partitionTypeBuilder.build();
     }
 
+    private Map<Integer, Type.PrimitiveType> buildIdPrimitiveTypeMapping(List<Types.NestedField> columns)
+    {
+        ImmutableMap.Builder<Integer, Type.PrimitiveType> idToTypeMapping = ImmutableMap.builder();
+        for (Types.NestedField field : columns) {
+            buildIdPrimitiveTypeMapping(field, idToTypeMapping);
+        }
+        return idToTypeMapping.build();
+    }
+
+    private void buildIdPrimitiveTypeMapping(Types.NestedField field, ImmutableMap.Builder<Integer, Type.PrimitiveType> icebergIdToTypeMapping)
+    {
+        Type type = field.type();
+        if (type.isPrimitiveType()) {
+            icebergIdToTypeMapping.put(field.fieldId(), type.asPrimitiveType());
+            return;
+        }
+        if (type instanceof Type.NestedType) {
```

Comment:
Sorry, I'm not seeing how these changes are related. This has to do with statistics for ROW type sub-columns?

Reply:
Ah, I see, there were two issues. Can you add the example from https://github.com/trinodb/trino/issues/7502 to the tests?

Reply:
I do not understand:
* which bug this commit is supposed to fix
* how the fix here relate to added test. The test does not seem to use nested types.

Reply:
> * how the fix here relate to added test.

indeed, the `testPartitionedTableWithDateTransformStatistics` test passes when run without other changes.